packages/ice/src/turn/protocol.ts

```diff
@@ -201,14 +213,24 @@ class TurnClient implements Protocol {
   refresh = () =>
     new PCancelable(async (r, f, onCancel) => {
       let run = true;
+      let timeoutHandle: NodeJS.Timeout | undefined;
       onCancel(() => {
         run = false;
+        if (timeoutHandle !== undefined) {
+          clearTimeout(timeoutHandle);
+        }
         f("cancel");
       });
 
       while (run) {
         // refresh before expire
-        await setTimeout((5 / 6) * this.lifetime * 1000);
+        let resolve: (value: void | PromiseLike<void>) => void = () => {};
+        const deferred = new Promise<void>((_r) => {
```

Comment:
nit: this could probably be simplified in to a single `new Promise` statement

Reply:
Done